### ISSUE TITLE
[Bug]: cagg migration crash

### ISSUE THREAD

**Issue by Xima** [timescale/timescaledb#4913]
### What type of bug is this?

Unexpected error

### What subsystems and features are affected?

Continuous aggregate

### What happened?

I tried to run the cagg migration and it failed after a few seconds.

The cagg was originally created on an old timescaledb v1.x

### TimescaleDB version affected

2.8.1

### PostgreSQL version used

14.5

### What operating system did you use?

Ubuntu 18.04

### What installation method did you use?

Deb/Apt

### What platform did you run on?

On prem/Self-hosted

### Relevant log output and stack trace

```bash
2022-11-02 14:31:08.162 CET [475985] LOG:  refreshing continuous aggregate "aggregated_test_table_pt1h" in window [ 4714-11-24 00:00:00+00 BC, 2022-11-02 11:00:00+00 ]
2022-11-02 14:31:37.249 CET [468989] postgres@historian_data ERROR:  portal snapshots (1) did not account for all active snapshots (2)
2022-11-02 14:31:37.249 CET [468989] postgres@historian_data CONTEXT:  SQL statement "CALL public.refresh_continuous_aggregate(_cagg, (_plan_step.config->>'window_start')::timestamptz, NULL)"
        PL/pgSQL function _timescaledb_internal.cagg_migrate_execute_refresh_new_cagg(_timescaledb_catalog.continuous_agg,_timescaledb_catalog.continuous_agg_migrate_plan_step) line 9 at CALL
        SQL statement "CALL _timescaledb_internal.cagg_migrate_execute_refresh_new_cagg($1, $2)"
        PL/pgSQL function _timescaledb_internal.cagg_migrate_execute_plan(_timescaledb_catalog.continuous_agg) line 31 at EXECUTE
        SQL statement "CALL _timescaledb_internal.cagg_migrate_execute_plan(_cagg_data)"
        PL/pgSQL function cagg_migrate(regclass,boolean,boolean) line 25 at CALL
2022-11-02 14:31:37.249 CET [468989] postgres@historian_data STATEMENT:  call cagg_migrate('aggregated_test_table_pt1h');
```

### How can we reproduce the bug?

```bash
I tried to recreate similar table/cagg with exactly same structure and add some fake data. But was not able to reproduce the bug. But able to reproduce it, if I recreate an "old" cagg on the same "production" hypertable.
```

**Comment by k1th** [comment]
same issue for us on PG 12.12

**Comment by k1th** [comment]
also, the crashed migration leaves orphaned data in
`_timescaledb_catalog.continuous_agg_migrate_plan_step`and `_timescaledb_catalog.continuous_agg_migrate_plan`
which will prevent retrying the migration later.

Please advise on how to clean this up…

**Comment by fabriziomello** [comment]
> also, the crashed migration leaves orphaned data in `_timescaledb_catalog.continuous_agg_migrate_plan_step`and `_timescaledb_catalog.continuous_agg_migrate_plan` which will prevent retrying the migration later.
> 
> Please advise on how to clean this up…

This bug was fixed in the PR #4898 and will be part of the next release.

To cleanup the data you can use execute DELETE in the catalog tables.

**Comment by k1th** [comment]
I can reproduce this with creating a cont agg with finalized=false like this (had to scramble some column names).
The base hypertable has been in the system since pre-2.0.
I recreated the setup with a brand new hypertable – same thing.

Were are on PG 12.12 with extension 2.8.1

```
create materialized VIEW xyz_test (
    t,
    …
) with (timescaledb.continuous, timescaledb.materialized_only = true, timescaledb.finalized=false)
AS SELECT 
    time_bucket('1 day', t) as t_bucket,
    …
    sum(p) AS parts_sum
   FROM xyzxyz
  GROUP BY time_bucket('1 day', t), …
with no data;

CALL refresh_continuous_aggregate('xyz_test', '2022-10-01', '2022-11-05');

CALL cagg_migrate('xyz_test', true, true);
```

**Comment by konskov** [comment]
Thank you @k1th for providing reproduction steps, I could reproduce with those. Here is the complete reproduction example that worked for me 
```
CREATE TABLE conditions (
  time timestamptz NOT NULL,
  location TEXT NOT NULL,
  location2 char(10) NOT NULL,
  temperature DOUBLE PRECISION NULL,
  humidity DOUBLE PRECISION NULL
);

select create_hypertable('conditions', 'time');
-- insert some data into it
INSERT INTO conditions (time, location, location2, temperature, humidity)
SELECT generate_series('2021-09-01 00:00+00'::timestamptz, '2021-11-20 00:00+00'::timestamptz, '1 day') as time, 'NYC', 'nycity', 40, 40;

-- contagg
CREATE MATERIALIZED VIEW conditions_summary_daily
WITH (timescaledb.continuous, timescaledb.materialized_only = true, timescaledb.finalized=false) AS
SELECT location,
   time_bucket(INTERVAL '1 day', time) AS bucket,
   AVG(temperature),
   MAX(temperature),
   MIN(temperature)
FROM conditions
GROUP BY location, bucket
WITH NO DATA;

CALL refresh_continuous_aggregate('conditions_summary_daily', '2022-10-01', '2022-11-05');

CALL cagg_migrate('conditions_summary_daily', true, true);
```

**Comment by alex-tate** [comment]
> > also, the crashed migration leaves orphaned data in `_timescaledb_catalog.continuous_agg_migrate_plan_step`and `_timescaledb_catalog.continuous_agg_migrate_plan` which will prevent retrying the migration later.
> > Please advise on how to clean this up…
> 
> This bug was fixed in the PR #4898 and will be part of the next release.
> 
> To cleanup the data you can use execute DELETE in the catalog tables.

I see this bug will be addressed in a future release but I'm just running into it now in v2.8.1 (on a single test cagg). In addition to adding entries to the migrate_plan catalog tables, some of the plan steps appear to have completed (SAVE WATERMARK, CREATE NEW CAGG, DISABLE POLICIES) before the cagg migration script bails out. Other than deleting the empty 'cagg_name'_new, should I worry about any of the other completed steps?
